allensdk/brain_observatory/behavior/sync/__init__.py

```diff
@@ -234,67 +234,3 @@ def get_sync_data(
         'stim_photodiode': get_stim_photodiode(sync_dataset, permissive),
         'stimulus_times_no_delay': get_raw_stimulus_frames(sync_dataset, permissive)
     }
-
-
-def frame_time_offset(data: Dict[str, Any]) -> float:
-    """
-    Contained in the behavior "pickle" file is a series of time between
-    consecutive vsync frames (`intervalsms`). This information required
-    to get the timestamp (via frame number) for events that occured
-    outside of a trial(e.g. licks). However, we don't have the value
-    in the trial log time stream when the first vsync frame actually
-    occured -- so we estimate it with a linear regression (frame
-    number x time). All trials in the `trial_log` have events for
-    `trial_start` and `trial_end`, so these are used to fit the
-    regression. A linear regression is used rather than just
-    subtracting the time from the first trial, since there can be some
-    jitter given the 60Hz refresh rate.
-
-    Parameters
-    ----------
-    data: dict
-        behavior pickle well-known file data
-
-    Returns
-    -------
-    float
-        Time offset to add to the vsync stream to sync it with the
-        `trial_log` time stream. The "zero-th" frame time.
-    """
-    events = [trial["events"] for trial
-              in data["items"]["behavior"]["trial_log"]]
-    # First event in `events` is `trial_start`, and last is `trial_end`
-    # Event log has following schema:
-    #    event name, description: 'enter'/'exit'/'', time, frame number
-    # We want last two fields for first and last trial event
-    trial_by_frame = list(chain(
-        [event[i][-2:] for event in events for i in [0, -1]]))
-    times = [trials[0] for trials in trial_by_frame]
-    frames = [trials[1] for trials in trial_by_frame]
-    time_to_first_vsync = sps.linregress(frames, times).intercept
-    return time_to_first_vsync
-
-
-def get_stimulus_rebase_function(data, stimulus_timestamps_no_monitor_delay):
```

Comment:
Can we get some context for why the `frame_time_offset` and `get_stimulus_rebase_function` are now being nixed in favor of simple 1 timestamp in pkl file == 1 timestamp in sync file line label?

I recall Kat spent a lot of time working with stakeholders to try get linear regression and time offsets working. Do stakeholders no longer want this anymore?

Reply:
The stakeholders decided they did not want to try to interpolate timestamps that occur between frame transitions and just wanted to assign events the timestamps of their corresponding frames. That conversation actually occurred in the bottom 2/3 of ticket 1818

https://app.zenhub.com/workspaces/allensdk-10-5c17f74db59cfb36f158db8c/issues/alleninstitute/allensdk/1818

the relevant chain of comments starts here

https://app.zenhub.com/workspaces/allensdk-10-5c17f74db59cfb36f158db8c/issues/alleninstitute/allensdk/1818#issuecomment-776323709

In the case of BehaviorOphys sessions, we have the sync time array which we can treat as ground truth, so `frame_time_offset` was not needed.

In the case of behavior-only sessions, VBA just asserts that the first frame is at time = 0. The stakeholders trust VBA, and want the SDK to mimic that behavior.